pkg/service/cluster.go

```diff
@@ -718,6 +720,9 @@ func (cs *clusterService) List(ctx context.Context, opts ...query.Option) (*infr
 	var proj models.Project
 	_, err := dao.GetByName(ctx, cs.db, queryOptions.Project, &proj)
 	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, status.Error(codes.NotFound, fmt.Sprintf("Project %v not found.", queryOptions.GetProject()))
```

Comment:
Can you change error message to something generic like `cluster not found` or `requested resource not found`?

A less privileged user (for example, project or namespace read only) who cannot retieve full projects list can determine existence of project with the help of error message.

Reply:
There are places where we return project name in error message which need to removed too. Let's take it separately.
cc @niravparikh05

Reply:
Update PR with new error message. Also added a test case to cover non-exists project scenario.